File: 0232-implement-queue-using-stacks/0232-implement-queue-using-stacks.py
class MyQueue:
    def __init__(self):
        self.list1 = []
        

    def push(self, x: int) -> None:
        self.list1.append(x)

    def pop(self) -> int:
        top  = self.list1[0]
        self.list1.pop(0)
        return top

    def peek(self) -> int:
        return self.list1[0]
        

    def empty(self) -> bool:
        return len(self.list1)==0
        


# Your MyQueue object will be instantiated and called as such:
# obj = MyQueue()
# obj.push(x)
# param_2 = obj.pop()
# param_3 = obj.peek()
# param_4 = obj.empty()

File: 0232-implement-queue-using-stacks/0232-implement-queue-using-stacks.cpp
class MyQueue {
public:
    stack<int> s1, s2;
    MyQueue() {}

    void push(int x) {
       
        s1.push(x);
    }

    int pop() {
        if (empty()) return -1;

        
        if (s2.empty()) {
            while (!s1.empty()) {
                s2.push(s1.top());
                s1.pop();
            }
        }

        int ele = s2.top();
        s2.pop();
        return ele;
    }

    int peek() {
        if (empty()) return -1;

        
        if (s2.empty()) {
            while (!s1.empty()) {
                s2.push(s1.top());
                s1.pop();
            }
        }

        return s2.top();
    }

    bool empty() {
        return s1.empty() && s2.empty();
    }
};
